src/main/java/com/bpodgursky/jbool_expressions/ExprUtil.java

```diff
@@ -29,26 +29,44 @@
     return Or.of(newChildren);
   }
 
-  public static <K> Expression<K> collapseToPOS(Or<K> or, And<K> internalAnd, Expression<K> omitFromAnd){
+  public static <K> Expression<K> stripNegation(And<K> and, Or<K> internalOr, Expression<K> omitFromOr){
```

Comment:
So I can see why you wrote this method; it feels strange to me to flip from SOP to POS.  I thought quite a bit about making this fail and I couldn't see anything.  But I think this could be simplified considerably (like having childrenNew and newChildren variables is just confusing; I know you copied that from the collapseToPOS method but).  Personally I would just write this out in full:

for (Expression child : and.expressions)
{
    if (newChildren.equals(internalOr))
    {
         List<Expression> newInternalOrChildren;
 
         etc
    }
    else 
       newChildren.add(child)
}

Reply:
This is from merging in the main project